geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java

```diff
@@ -201,15 +200,24 @@ public void init(Services<ID> s) throws MembershipConfigurationException {
     boolean enableNetworkPartitionDetection = config.getEnableNetworkPartitionDetection();
     System.setProperty("jgroups.resolve_dns", String.valueOf(!enableNetworkPartitionDetection));
 
-    InputStream is;
+    InputStream is = null;
 
     String r;
     if (config.isMcastEnabled()) {
       r = JGROUPS_MCAST_CONFIG_FILE_NAME;
     } else {
       r = DEFAULT_JGROUPS_TCP_CONFIG;
     }
-    is = ClassPathLoader.getLatest().getResourceAsStream(getClass(), r);
+    ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+    if (contextClassLoader != null) {
+      is = contextClassLoader.getResourceAsStream(r);
+    }
```

Comment:
I see the new code searches first, using the current thread's context `ClassLoader`, and then tries the current class' one. Finally it searches for the resource as a "system resource" (in the classpath).

This seems fine, in the abstract. But I can't tell how it relates to the old code `ClassPathLoader.getLatest().getResourceAsStream()`.

Reply:
ClassPathLoader exists to find customer classes in multiple contexts including J2EE servers, Web servers, regular gfsh-launched servers, etc.  Here we only need to find the JGroups configuration file, which will normally be found with ClassLoader.getSystemResourceAsStream(r).  I added the others because you really should look at the thread-context class-loader first and then, why not look at the class-loader for JGroupsMessenger as well (that's where I thought it would be found but I was wrong for the common case).

Reply:
thanks @bschuchardt